app/src/main/java/hypr/a255bits/com/hypr/ModelFragmnt/ModelInteractor.kt

```diff
@@ -21,6 +26,30 @@ class ModelInteractor(val context: Context) : ModelFragmentMVP.interactor {
                 .build()
     }
 
+    override fun getIntentForSharingImagesWithOtherApps(imageFromGallery: Bitmap?): Intent {
+        val imageInBytes = imageFromGallery?.let { BitmapManipulator().compressBitmapToByteArray(Bitmap.CompressFormat.PNG, it) }
+
+        val tempStorageFileLocation = Environment.getExternalStorageDirectory().absolutePath + File.separator + "temporary_file.jpg"
+        writeByteArrayToFile(tempStorageFileLocation, imageInBytes)
+
+        return createImageShareIntent(Uri.parse("file:///sdcard/temporary_file.jpg"))
```

Comment:
This seems good for now.  If the user doesn't have an SDcard we can probably still find somewhere to store things with something like:

https://stackoverflow.com/questions/3425906/creating-temporary-files-in-android